lib/Extended_ast.ml

```diff
@@ -101,6 +107,26 @@ module Parse = struct
 
   let normalize fg x = map fg fix_letop_locs @@ map fg normalize_lists @@ x
 
+  let repl_file (lx : Lexing.lexbuf) =
+    let str = String.strip (Bytes.to_string lx.lex_buffer) in
+    let phrases = Astring.String.cuts ~sep:"\n# " str in
+    let phrases =
+      List.mapi phrases ~f:(fun i p ->
+          let p = String.strip p in
+          match i with
+          | 0 -> Option.value (String.chop_prefix p ~prefix:"# ") ~default:p
+          | _ -> p )
+    in
+    List.map phrases ~f:(fun p ->
+        let p, prepl_output =
+          match Astring.String.cut ~sep:";;\n" p with
+          | Some (p, o) -> (p ^ ";;", o)
+          | None -> (p, "")
```

Comment:
If the rule above is not implemented, this isn't a good default to return. The top-level output part will certainly cause a syntax error.

Reply:
If an exception is raised when parsing as a toplevel_phrase, the code block will be printed verbatim, which is the least surprising outcome for the user I think. If there is no `;;` then we assume that the whole input is the toplevel_phrase, we can't really guess if there is the output in the block and where it starts.

Reply:
I think it's better to raise an exception at this point and to not attempt to parse.

Reply:
I replied in the other comment, I think it's better to mimic the other standard parsing functions (thus raising exceptions from Syntaxerr), all exceptions are caught anyway and the intput = output in this case (we don't do anything specific in the error case)